forte/data/extractor/predictor.py

```diff
@@ -0,0 +1,199 @@
+                                },
+                                "ner_tag": {
+                                    "extractor":  extractor2,
+                                    "converter": converter2
+                                }
+                            }
+
+        class Model:
+            def __call__(self, tensor):
+                # Input shape: [Batch, length]
+                # Output shape: [Batch]
+                batch_size = len(tensor)
+                length = len(tensor[0])
+                return [[2 for _ in range(length)] for _ in range(batch_size)]
+        self.model = Model()
+
+    def convert(self, feature_collections):
+        example_collection = {}
+        for features_collection in feature_collections:
+            for tag, feature in features_collection.items():
+                if tag not in example_collection:
+                    example_collection[tag] = []
+                example_collection[tag].append(feature)
+
+        tensor_collection = {}
+        for tag, features in example_collection.items():
+            converter = self.feature_scheme[tag]["converter"]
+            tensor, mask = converter.convert(features)
+
+            if tag not in tensor_collection:
+                tensor_collection[tag] = {}
+
+            tensor_collection[tag]["tensor"] = tensor
+            tensor_collection[tag]["mask"] = mask
+        return tensor_collection
+
+    def yield_batch(self, pack):
+        for instance in pack.get(self.scope):
+            feature_collection = {}
+            for tag, scheme in self.feature_scheme.items():
+                extractor = scheme["extractor"]
+                feature = extractor.extract(pack, instance)
+                feature_collection[tag] = feature
+            
+            self.pack_pools.append(pack)
+            self.instance_pools.append(instance)
+            self.features_pools.append(feature_collection)
+
+            if len(self.features_pools) == self.batch_size:
+                yield self.convert(self.features_pools[:self.batch_size]), \
+                    self.pack_pools[:self.batch_size], \
+                    self.instance_pools[:self.batch_size]
+                
+                self.features_pools.clear()
+                self.pack_pools.clear()
+                self.instance_pools.clear()
+
+    def flush_batch(self):
+        if len(self.features_pools) > 0:
+            yield self.convert(self.features_pools), self.pack_pools, \
+                self.instance_pools
+            self.features_pools.clear()
+            self.pack_pools.clear()
+            self.instance_pools.clear()
+
+    def _process(self, input_pack: PackType):
+        for tensor_collection, packs, instances in self.yield_batch(input_pack):
+            predictions = self.predict(tensor_collection)
+            self.add_to_pack(predictions, tensor_collection, packs, instances)
+
+        # update the status of the jobs. The jobs which were removed from
```

Comment:
Did you copy this part from some other class?

Reply:
The original predictor has functions called "predict" and "pack", basically doing the same things here. One is to get the output, the other one is to add output to pack.